pydicom/jsonrep.py

```diff
@@ -4,194 +4,278 @@
+                x = cast(Callable[[str], bytes], handler)
                 return x(value)
 
             self.bulk_data_element_handler = wrapper
         else:
-            handler = cast(Callable[[BaseTag, str, str], Any], handler)
-            self.bulk_data_element_handler = handler
+            self.bulk_data_element_handler = cast(BulkDataHandlerType, handler)
 
-    def get_element_values(self):
+    def get_element_values(self) -> Any:
         """Return a the data element value or list of values.
 
         Returns
         -------
-        str or bytes or int or float or dataset_class
-        or PersonName or list of any of these types
+        Any
             The value or value list of the newly created data element.
         """
         from pydicom.dataelem import empty_value_for_VR
+
+        if self.value_key is None:
+            return empty_value_for_VR(self.vr)
+
+        # An attribute with an empty value shall have no "Value", "BulkDataURI"
+        #   or "InlineBinary"
         if self.value_key == 'Value':
+            # May be a single item or a list of items
+            # A multi-valued value uses `null` for empty items
+            # An empty sequence is `{}`
             if not isinstance(self.value, list):
-                fmt = '"{}" of data element "{}" must be a list.'
-                raise TypeError(fmt.format(self.value_key, self.tag))
+                raise TypeError(
+                    f"'{self.value_key}' of data element '{self.tag}' must "
+                    "be a list"
+                )
+
             if not self.value:
                 return empty_value_for_VR(self.vr)
-            element_value = [self.get_regular_element_value(v)
-                             for v in self.value]
+
+            val = cast(List[ValueType], self.value)
+            element_value = [self.get_regular_element_value(v) for v in val]
             if len(element_value) == 1 and self.vr != 'SQ':
                 element_value = element_value[0]
+
             return convert_to_python_number(element_value, self.vr)
 
         # The value for "InlineBinary" shall be encoded as a base64 encoded
         # string, as shown in PS3.18, Table F.3.1-1, but the example in
         # PS3.18, Annex F.4 shows the string enclosed in a list.
         # We support both variants, as the standard is ambiguous here,
         # and do the same for "BulkDataURI".
-        value = self.value
+        value = cast(Union[str, List[str]], self.value)
         if isinstance(value, list):
             value = value[0]
 
         if self.value_key == 'InlineBinary':
-            if not isinstance(value, (str, bytes)):
-                fmt = '"{}" of data element "{}" must be a bytes-like object.'
-                raise TypeError(fmt.format(self.value_key, self.tag))
-            return base64.b64decode(value)
+            # The `value` should be a base64 encoded str
+            if not isinstance(value, str):
```

Comment:
According to [this](https://docs.python.org/3/library/json.html#json-to-py-table), by default Python doesn't convert JSON to bytes